adafruit_platformdetect/board.py

```diff
@@ -151,6 +151,8 @@ def id(self) -> Optional[str]:
             board_id = self._udoo_id()
         elif chip_id == chips.STM32MP157:
             board_id = self._stm32mp1_id()
+        elif chip_id == chips.STM32MP157AAA3:
+            board_id = self._stm32mp157_id()
```

Comment:
You should just be able to use `self._stm32mp1_id()` so the id code isn't so scattered.

Reply:
i have already thought about that, but:
For the next step in adding the new board i noticed i needed to distinct between two chip classes:
The small 12x12 package that everyone else uses (like on the STM32MP157-DK2, and every other board already implemnented here) and
The big 18x18 package that is used by olimex

later in the process i noticed that i needed to add another pin and board file since the already existing pin mapping is not compatible.
Therefore i have further added the above section in the code to distinct between the small TFBGA361 12x12 package and the big LFBGA447 18x18 package